R/conditions.R

```diff
@@ -359,10 +359,28 @@ printStackTrace <- function(cond,
     list(attr(cond, "stack.trace", exact = TRUE))
   )
 
+  # Stripping of stack traces is the one step where the different stack traces
+  # interact.
+  stripResults <- stripStackTraces(lapply(stackTraces, function(trace) {
+    if (is.integer(trace)) {
+      # This is where deep stacks were elided.
+      # jcheng 2024-12-03: For the purposes of stack trace stripping, it's not
+      # ideal that we're eliding entire stacks before we get here--this actually
+      # potentially screws up our scoring (i.e., an unbalanced ..stacktraceon..
+      # or ..stacktraceoff.. was elided). If this surfaces as an actual problem,
+      # we could maybe maintain a cumulative score of the stacks as we elide
+      # them.
+      character(0)
+    } else {
```

Comment:
```suggestion
    } else if (!full) {
      character(0)
    } else {
```

Reply:
I handled this slightly differently so we don't go through the `stripStackTraces` code path at all if `full` is true.